.github/workflows/branch-sync.yml

```diff
@@ -0,0 +1,85 @@
+name: Merge 6.x into 7.x
+
+on:
+  schedule:
+    - cron: '0 23 * * 0'  # Run every Sunday at 11 PM UTC (Monday 9 AM AEST)
+
+permissions:
+  contents: write
+  pull-requests: write
+
+jobs:
+  create-pull-request:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Configure git
+        run: |
+          git config user.name "github-actions[bot]"
+          git config user.email "github-actions[bot]@users.noreply.github.com"
+
+      - name: Prepare sync branch
+        run: |
+          # Fetch all branches
+          git fetch origin
+
+          # Create a new sync branch from 7.x (target)
+          git checkout -b temp-sync-branch origin/7.x
+          
+          # Cherry-pick or merge changes from 6.x that are not in 7.x
+          git checkout origin/6.x -- .
+          
+          # Check if there are any changes
+          if [[ -z $(git status --porcelain) ]]; then
+            echo "NO_CHANGES=true" >> "${GITHUB_ENV}"
+            exit 0
+          fi
+          
+          # Commit the changes to our temporary branch
+          git add .
+          git commit -m "Sync changes from 6.x to 7.x"
+          
+          # Push the sync branch
+          git push origin temp-sync-branch --force
+
+      - name: Check for conflicts
+        if: env.NO_CHANGES != 'true'
+        id: check_conflicts
+        run: |
+          # Test if this would create conflicts with 7.x
+          if git diff --check; then
+            echo "has_conflicts=false" >> "${GITHUB_OUTPUT}"
+          else
+            echo "has_conflicts=true" >> "${GITHUB_OUTPUT}"
+          fi
+
+      - name: Create a pull request from sync branch to 7.x
+        if: env.NO_CHANGES != 'true' && steps.check_conflicts.outputs.has_conflicts == 'false'
+        uses: peter-evans/create-pull-request@v7
```

Comment:
When reviewing this action I saw this feature for [commit signing for bots](https://github.com/peter-evans/create-pull-request/blob/main/docs/concepts-guidelines.md#commit-signature-verification-for-bots) which sounds simple enough to implement when using the injected `GITHUB_TOKEN`. 

It's not necessary because we don't enforce signed commits for this repo but I like the idea of it being implemented to cover prepare for a future state where we do enforce it.

Reply:
lets give it a crack!